build_tools/scripts/update_submodules.py

```diff
@@ -38,18 +38,23 @@
 
 
 def parse_arguments():
-  default_repo_path = os.path.dirname(
-      os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
   parser = argparse.ArgumentParser()
   parser.add_argument(
-      "--let_met_through",
-      help="This tool is not ready yet. Please use update_submodules.sh for now"
-  )
+      "--let_me_through",
+      help="This tool is not ready yet. Please use update_submodules.sh "
+      "for now",
+      type=str2bool)
+  parser.add_argument("--repo", help="Repository root directory")
   parser.add_argument(
-      "--repo", help="Repository root directory", default=default_repo_path)
+      "--tensorflow",
+      help="Path to the tensorflow sources "
+      "(default to third_party/tensorflow)",
```

Comment:
This implies that it's relative to the repo root, but I think it has to be absolute. Can you make it relative to the repo root instead? Then this would be correct and I think it's just generally better. Plus then you could just give this a default here and I assume argparse will generate the appropriate help text for you.

Reply:
It is defaulting it relative to the repo root but manages that below. I think the two step None default is required. Otherwise, if I overrode the repo root via the --repo flag, these would still be anchored to the old default. Annoying but I've written this pattern many times and not found a less opaque way to do it.

Reply:
Right it defaults as this path relative to the repo root, but it only prepends the repo root if this arg isn't set, so I think if I passed `--tensorflow=my_tensorflow`, it would not override it anywhere and use that as an absolute path (unless the cwd args are magically relative to something?). I think the documentation here implies that I can pass it a relative path. I think what you could do is set `default=third_party/tensorflow` here and then unconditionally prepend args.repo below (instead of only if this is unset). Then if I pass some other path here, it becomes relative to the repo root.

Reply:
Unless you *want* the option to set this as an absolute path, in which case maybe just add "[repo]" at the beginning here to indicate that the default is relative but folks need to write out the whole path if they're overriding. 

There is probably a way to make the path joining libraries treat trailing absolute paths specially as always absolute rather than blinding concatenating (at least one of the C++ libraries we use does it that way), but seems like more trouble than it's worth right now.